### DISCUSSION TITLE
[Q] Macro/VBAScript injection in a new Excel file

### DISCUSSION THREAD

**Discussion by adnaanbheda** [exceljs/exceljs#2421] [Q&A]
## 💬 Questions and Help

Hello, I'm looking into creating a custom excel file that incorporates VBAScripts for data validation.

My use case includes creating a template VBA Script and updating the variables inside that script so that they can be used as a dropdown list in an excel cell.

If Exceljs doesn't support this, suggestions and approaches to this problem are welcome ! 

Cheers.

**Comment by buckwheat24** [comment]
+1 - I would also love to be able to add VBA code to the Excel file. Ultimately my team would like to have a dropdown where we can select multiple values from the list and this can be achieved via VBA.

**Comment by adnaanbheda** [comment]
@buckwheat24 we ended up using data validation formulaes that are available in this library, using multiple hidden sheets that contain the dropdown list data.

**Comment by radiumrasheed** [comment]
This works but I'm trying to make the dropdowns dynamic, so based on the value of one dropdown, we display other dropdowns. anyway to achieve this?

**Comment by radiumrasheed** [comment]
For me, I used the INDIRECT formula with defined names to achieve cascading dropdown

**Comment by adnaanbheda** [comment]
@radiumrasheed We used a combination of INDEX and MATCH to achieve the same result :) cheers, although it would've been nice to get the support for macros ! 😅
